### ISSUE TITLE
Remove minor memory leak from navigation

### ISSUE THREAD

**Issue by inf3rno** [inf3rno/e2e#61]
By load I do the following:

```js
load(resourceIdentifier) {
	return new Promise((resolve, reject) => {
		this.once("page:complete", () => {
			resolve(this.window);
		});
		this.once("navigation:error", reject);
		this.window.document.location.href = resourceIdentifier;
	});
}
```

The only problem with this approach, that it keeps the event listeners. So if the complete wins, then it does not clear the once listener for the navigation error. If the navigation error wins, then it does not clear the once listener for the completion. I am not sure whether it was a wise idea to solve this with event listener. I added them because this way it is easier to develop tests with more complex navigation than just did load / did not load. For example load is readyState:complete now, but for most of the tests a readyState:interactive is just enough... I think I should support both somehow, but I don't know how I can do so. I mean it would be strange to return many promises instead of just one. Another issue that what should happen by page error? The navigation error means rejection, but the page error means nothing from the WindowManager perspective. It is just an event. It does not really affect navigation, but it affects the tests after the navigation is complete. So in most of the cases a page error means failure.

**Comment by inf3rno** [comment]
A possible solution might be to call the win.load with more params, e.g. 

```js
win.load("x.html", {resolve: ["page:interactive"], reject: ["navigation:error", "page:error"]}
```

Ofc. we could have default values we could set for example by instantiation. The default of the lib would be the upper, but an user of the lib would be able to change it. The interactive, page error, etc. means event + next tick here, otherwise we would resolve or reject the promise before the other listeners are called. That can be a problem if we test a single page application and we have to wait for some kind of init script. The "page:interactive" would run before the init script, because we added that before the script of the loaded page ran and added the listener to the readystatechange event. I think the upper should contain something like "after:page:interactive" and "before:page:interactive" or something like that to make sure that the resolve or reject is called at the right time.

As of the other events; page logging should be collected in a testable log, e.g. `expect(log.contains("blah")).to.equal(true)`. So logging should not be an event I think. On the other hand we could await for logs. So I think we should think more about this part of the API.

How to fix the once issue?

Well I think we should add a clearing procedure before resolving or rejecting the promise. We already have the lists of the events we are listening to, so it is relative easy to remove all of them. They won't make a real problem though, because the promise can be resolved/rejected only once. But it is a memory leak...

**Comment by inf3rno** [comment]
I delay this to v0.2 to release earlier.

**Comment by inf3rno** [comment]
What we really need here is somehow binding these event handlers together, so we will be able to remove them if any of them is called. I think we need to wrap them for that and the wrapping should remove all of them before one of them is called. This is not that hard. I think we need an utility function for it, and we need to test that function just to be sure.

**Comment by inf3rno** [comment]
I think I solve this next. It is an old bug.
